### PR TITLE
Prevent two contracts from having the same name

### PR DESCRIPTION
**Pull request by lcswillems**
Currently, two contracts of the project can have the same name. In this case, the contract that appears the latest in the tree view will be the one present in the `artifacts.json`. This is a dangerous behavior. The simplest way is to prevent two contracts from having the same name.

### CODE REVIEW

File: multiversx_sdk_rust_contract_builder/builder.py [hunk 1]
```diff
@@ -92,12 +93,21 @@ def ensure_output_folder_is_empty(parent_output_folder: Path):
         raise ErrKnown(f"Output folder must be empty: {parent_output_folder}")
 
 
-def get_contracts_folders(project_path: Path) -> List[Path]:
+def get_contracts(project_path: Path):
```
Comment (andreibancioiu) [inline multiversx_sdk_rust_contract_builder/builder.py:96]: Perhaps keep this function as it was, and apply the check in a separate function `ensure_distinct_contract_names`.

File: multiversx_sdk_rust_contract_builder/builder.py [hunk 2]
```diff
@@ -39,16 +39,17 @@ def build_project(
     ensure_output_folder_is_empty(parent_output_folder)
 
     outcome = BuildOutcome(metadata, options)
-    contracts_folders = get_contracts_folders(project_folder)
+    contracts = get_contracts(project_folder)
```
Comment (andreibancioiu) [inline multiversx_sdk_rust_contract_builder/builder.py:42]: Perhaps keep this as it were, then add a check below it:

```
ensure_distinct_contract_names(contracts_folders)
```

Which makes use of `get_contract_name_and_version`.

The validation logic would be a bit more explicit this way.

File: multiversx_sdk_rust_contract_builder/builder.py (hunk 1 above)
Comment (lcswillems) [reply multiversx_sdk_rust_contract_builder/builder.py:96]: `ensure_distinct_contract_names` would duplicate the logic inside the for loop.

Would it be fine if check is done at the beginning of the for loop using a set `contract_names`?

Comment (andreibancioiu) [reply multiversx_sdk_rust_contract_builder/builder.py:96]: In this context, I think it's fine to have a loop and the calls to `get_contract_name_and_version` in `ensure_distinct_contract_names` (even if those exist in the main loop as well) - not duplication per-se.

Yes, using a set should also be all right.

**Review by andreibancioiu — DISMISSED** [review]

**Review by schimih — APPROVED** [review]

**Pull request merged**